packages/tradershub/src/flows/RealAccountCreation/WizardScreens/WizardScreens.tsx

```diff
@@ -12,19 +20,20 @@ import { Address, CurrencySelector, PersonalDetails, TermsOfUse } from '@/screen
  */
 const WizardScreens = () => {
     const { currentStep } = useRealAccountCreationContext();
+    const { isEU } = useRegulationFlags();
 
-    switch (currentStep) {
-        case 1:
-            return <CurrencySelector />;
-        case 2:
-            return <PersonalDetails />;
-        case 3:
-            return <Address />;
-        case 4:
-            return <TermsOfUse />;
-        default:
-            return null;
-    }
+    // If the user is in the EU, we need to display the TradingAssessment and FinancialAssessment screens
+    const screens = useMemo(() => {
+        if (isEU) {
+            return [CurrencySelector, PersonalDetails, Address, TradingAssessment, FinancialAssessment, TermsOfUse];
+        }
+        return [CurrencySelector, PersonalDetails, Address, TermsOfUse];
```

Comment:
let's create a hook for this, better to keep EU/nonEU decisions out of the components.

Reply:
Totally agree @matin-deriv. However these are the JSX screens and if we create a hook it'll be the same logic where we'll return the array of screens at most. And consuming them here. IMO it'll just be an additional hook in this scenario.